src/app/dialog/dialog.component.ts

```diff
@@ -0,0 +1,64 @@
+import {Component, OnInit} from '@angular/core';
```

Comment:
This component has generic aspects so that you can use it in multiple places, but it also has a ton of edge cases and coupling. Either make it generic with little to none of the coupling, or split it to multiple components